designer/report-designer/src/main/java/org/pentaho/reporting/designer/core/actions/elements/MorphAction.java

```diff
@@ -63,6 +66,9 @@
   private final ElementMetaData metaData;
   private ElementType targetElementType;
 
+  public static final String[] COMPLEX_ELEMENT_NAMES = new String[] { "band", "sub-report", "crosstab-report", "toc", "index" };
```

Comment:
Use the metadata! The ElementType#getTypeClassification tells you whether a element is a Subreport or a control-object. (Control objects are things like Groups, which cannot be morphed either.